src/Features/Core/Portable/EmbeddedLanguages/Json/JsonParser.cs

```diff
@@ -371,13 +371,25 @@ private bool ShouldConsumeSequenceElement()
             };
 
         private JsonValueNode ParseValue()
-            => _currentToken.Kind switch
+        {
+            try
             {
-                JsonKind.OpenBraceToken => ParseObject(),
-                JsonKind.OpenBracketToken => ParseArray(),
-                JsonKind.CommaToken => ParseCommaValue(),
-                _ => ParseLiteralOrPropertyOrConstructor(),
-            };
+                _recursionDepth++;
+                StackGuard.EnsureSufficientExecutionStack(_recursionDepth);
```

Comment:
we already do this recursion check in ParseSequence (which is called inside json arrays and objects).  The issue there was that hte parser has a piece of self-recursion in the parser (ParseValue) which doesn't have to go through that in a case like what the tests show.